src/components/CIPipelineN/CustomInputVariableSelect.tsx

```diff
@@ -0,0 +1,285 @@
+        formData,
+        setFormData,
+        selectedTaskIndex,
+        activeStageName,
+        inputVariablesListFromPrevStep,
+        globalVariables,
+    }: {
+        formData: FormType
+        setFormData: React.Dispatch<React.SetStateAction<FormType>>
+        selectedTaskIndex: number
+        activeStageName: string
+        inputVariablesListFromPrevStep: {
+            preBuildStage: Map<string, VariableType>[]
+            postBuildStage: Map<string, VariableType>[]
+        }
+        globalVariables: { label: string; value: string; format: string }[]
+    } = useContext(ciPipelineContext)
+    const [selectedOutputVariable, setSelectedOutputVariable] = useState<{
+        label: string
+        value: string
+    }>({
+        label: '',
+        value: '',
+    })
+
+    const [inputVariableOptions, setInputVariableOptions] = useState<
+        {
+            label: string
+            options: any[]
+        }[]
+    >([])
+
+    useEffect(() => {
+        const previousStepVariables = []
+        if (inputVariablesListFromPrevStep[activeStageName].length > 0) {
+            inputVariablesListFromPrevStep[activeStageName][selectedTaskIndex].forEach((element) => {
+                previousStepVariables.push({ ...element, label: element.name, value: element.name })
+            })
+        }
+        if (activeStageName === BuildStageVariable.PostBuild) {
+            const preBuildStageVariables = []
+            const preBuildTaskLength = formData[BuildStageVariable.PreBuild]?.steps?.length
+            if (preBuildTaskLength >= 1) {
+                if (inputVariablesListFromPrevStep[BuildStageVariable.PreBuild].length > 0) {
+                    inputVariablesListFromPrevStep[BuildStageVariable.PreBuild][preBuildTaskLength - 1].forEach(
+                        (element) => {
+                            preBuildStageVariables.push({ ...element, label: element.name, value: element.name })
+                        },
+                    )
+                }
+
+                const stepTypeVariable =
+                    formData[BuildStageVariable.PreBuild].steps[preBuildTaskLength - 1].stepType === PluginType.INLINE
+                        ? 'inlineStepDetail'
+                        : 'pluginRefStepDetail'
+                const outputVariablesLength =
+                    formData[BuildStageVariable.PreBuild].steps[preBuildTaskLength - 1][stepTypeVariable]
+                        ?.outputVariables?.length || 0
+                for (let j = 0; j < outputVariablesLength; j++) {
+                    if (
+                        formData[BuildStageVariable.PreBuild].steps[preBuildTaskLength - 1][stepTypeVariable]
+                            .outputVariables[j].name
+                    ) {
+                        const currentVariableDetails =
+                            formData[BuildStageVariable.PreBuild].steps[preBuildTaskLength - 1][stepTypeVariable]
+                                .outputVariables[j]
```

Comment:
same here - try to move `formData[BuildStageVariable.PreBuild].steps[preBuildTaskLength - 1]` to some var